### DISCUSSION TITLE
layout of code generated figures

### DISCUSSION THREAD

**Discussion by nitin-ppnp** [quarto-dev/quarto-cli#3521] [Q&A]
Hi, I am generating many figures (>100) and want to place them in two columns and multiple pages. 

If don't use any layout option, each page contains one figure resulting in pages equal to the number of figures. If I use `#| layout-ncol: 2` in the code block, quarto put all the figures in two columns but within a single page which results in figures going out of the page. Output files in both cases (no layout, 2 col layout) are attached. The code for each file is included in the file itself.

[case1.pdf](https://github.com/quarto-dev/quarto-cli/files/10123479/case1.pdf)
[case2.pdf](https://github.com/quarto-dev/quarto-cli/files/10123482/case2.pdf)

**Comment by mcanouil** [comment]
Beyond the fact that a document with 100 figures in two columns layout is definitely not something I would want to open.

Since Quarto can't compute figure metric size, LaTeX class metric size and guess when to add a page break, you'll have to do that yourself by writing code that produce markdown (or LaTEX) using the `#| output: asis`.

**Reply by nitin-ppnp** [reply]
Thanks for the reply. This was helpful, but I still couldn't get the desired results. I am setting figure layout as 2 columns, but it is coming out to be 2 rows. Please check the attached doc for code and output.
[temp.pdf](https://github.com/quarto-dev/quarto-cli/files/10142649/temp.pdf)

**Reply by mcanouil** [reply]
The PDF does not help.
What would help is a (small) reproducible example showing your issue (when sharing code cell chunk embed the whole quarto document code inside four backticks).

`````
````
---
title: bla
---

Text

```{r}
1+1
```
````
`````

**Reply by nitin-ppnp** [reply]
Sorry, here are the file contents.
````
---
execute:
  keep-md: true
---
```{python}
#| layout-ncol: 2
#| output: asis

import plotly.graph_objects as go
import numpy as np
from IPython.display import display, Markdown, Latex

for i in range(20):
    fig = go.Figure(layout={"height":300})
    fig.add_trace(go.Image(z=np.random.randint(0,255,(100,100,3))))
    fig.update_xaxes(visible=False)
    fig.update_yaxes(visible=False)
    if i%2==0:
        display(Markdown("::: {layout-ncol=2}"))
    fig.show(renderer="jpeg")
    if i%2==1:
        display(Markdown(":::"))
        display(Markdown("{{< pagebreak >}}"))

```
````

**Reply by mcanouil** [reply]
Sorry for not getting back sooner.
The issue is that there is no empty lines between the two figures leaving Quarto with one single paragraph which it can't split.

I went dirty by adding a lot of `\n`, but you can fine tune this.

````
---
format: pdf
keep-md: true
---

```{python}
#| layout-ncol: 2
#| output: asis

import plotly.graph_objects as go
import numpy as np
from IPython.display import display, Markdown, Latex

for i in range(20):
    fig = go.Figure(layout={"height":300})
    fig.add_trace(go.Image(z=np.random.randint(0,255,(100,100,3))))
    fig.update_xaxes(visible=False)
    fig.update_yaxes(visible=False)
    if i%2==0:
        display(Markdown("\n:::{layout-ncol=2}\n"))
    fig.show(renderer="jpeg")
    display(Markdown("uit
    \n"))
    if i%2==1:
        display(Markdown("\n:::\n"))
        display(Markdown("\n{{< pagebreak >}}\n"))
```

````

<img width="1156" alt="image" src="https://user-images.githubusercontent.com/8896044/216709976-cb35da97-3acd-4935-9da1-c7ebd330028e.png">
